### PR TITLE
[Merged by Bors] - feat (Algebra/Module/LinearMap/Defs): the identity map of the zero module is the zero map

### PR DESCRIPTION
**Pull request by matematiflo**
Add a lemma about the identity map of the zero module.

This contribution was created as part of the AIM workshop *Formalizing algebraic geometry* in June 2024.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/LinearMap/Defs.lean
Comment (riccardobrasca) [inline Mathlib/Algebra/Module/LinearMap/Defs.lean:995]:
```suggestion
@[simp]
theorem identityMapOfZeroModuleIsZero [Subsingleton M] : id (R := R₁) (M := M) = 0 :=
```

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

**Comment by riccardobrasca** [comment]
bors r-

What do you think about adding the `simp` tag?

File: Mathlib/Algebra/Module/LinearMap/Defs.lean
Comment (matematiflo) [reply Mathlib/Algebra/Module/LinearMap/Defs.lean:995]: Done! Thanks 👍🏻

**Comment by riccardobrasca** [comment]
@matematiflo don't worry if you are confused about `git`, but:
* the `simp` isn't added, I don't know whether you forgot to push or you messed up something in trying to apply my suggestion.
* in general, even if you get a delegation, please wait for CI (in practice wait for all the checks to be green) before merging.

But again, don't worry, the first time the process is a little confusing!

**Comment by riccardobrasca** [comment]
Note that since I `bors r-`, one check will stay red (this is a bug in github in my opinion). You can safely ignore that. But all the other checks should be green.

**Review by matematiflo — COMMENTED** [review]
Add `@[simp]` attribute. Will try `bors r+` again next.

**Comment by riccardobrasca** [comment]
> bors r+

immagine
You should wait to the yellow thing here to become green. :D

**Comment by riccardobrasca** [comment]
The rationale is the following: when you merge it, a bot will start working, but not only on your PR, it creates batches of several PRs at the same time. If your PR has a problem the whole batch will fail, and everybody will loose some time (nothing serious!). This is why we ask author to "wait for CI": this means that another bot has checked that your PR is OK.

bors r-

**Comment by riccardobrasca** [comment]
Anyway this means that our instructions are not clear, it's surely not your fault.

**Comment by riccardobrasca** [comment]
For example I suspect that a linter will complain saying that `simp can prove this, so it cannot be simp`. Of course this means that my suggestion wasn't accurate, but I am just lazy to test it by myself.

**Comment by matematiflo** [comment]
OK. What threw me off is the message from 1h ago saying "✌️ matematiflo can now approve this pull request. To approve and merge a pull request, simply reply with bors r+. More detailed instructions are available [here](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)." I simply tried to follow that instruction.

**Comment by riccardobrasca** [comment]
Yeah, that message is not accurate. I don't know whether it depends on us or on the bot. Let me stress that this is not a problem at all, even if you try to merge something that does not work at end, the worst can happen is that a computer waist some time, you cannot break mathlib.

**Comment by matematiflo** [comment]
I checked again and `simp` cannot solve this, so we're safe on that front. As for the rest of the merging process, I am available if further actions are required on my end.

**Comment by riccardobrasca** [comment]
bors merge

**Pull request closed without merging**
